.travis.yml

```diff
@@ -1,33 +1,72 @@
 os: linux
 dist: focal
-language: c++
-env:
-  global:
-       - DEBIAN_FRONTEND: noninteractive
-addons:
-  apt:
-    packages:
-        - wget
-        - locales
-        - python3-pip
+language: minimal
+
+services:
+  - docker
 
-before_install:
-        - sudo locale-gen "en_US.UTF-8"
-        - export LANG="en_US.UTF-8"
-        - pip install git+https://github.com/antmicro/tuttest
-        - tuttest README.md unnamed0 | bash -
-jobs:
-  include:
-    - stage: SymbiFlow
-      install:
-            - tuttest README.md unnamed1 | bash -
+env:
+  # Testing QuickLogic toolchain on all OSes
+  - TOOLCHAIN=quicklogic      OS=ubuntu OS_VERSION=trusty
+  - TOOLCHAIN=quicklogic      OS=ubuntu OS_VERSION=xenial
+  - TOOLCHAIN=quicklogic      OS=ubuntu OS_VERSION=bionic
+  - TOOLCHAIN=quicklogic      OS=ubuntu OS_VERSION=eoan
+  - TOOLCHAIN=quicklogic      OS=ubuntu OS_VERSION=focal
+  - TOOLCHAIN=quicklogic      OS=centos OS_VERSION=6
+  - TOOLCHAIN=quicklogic      OS=centos OS_VERSION=7
+  - TOOLCHAIN=quicklogic      OS=centos OS_VERSION=8
+  # Testing Xilinx 7 Series toolchain on all OSes
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=trusty
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=xenial
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=bionic
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=eoan
+  - TOOLCHAIN=xilinx-series-7 OS=ubuntu OS_VERSION=focal
+  - TOOLCHAIN=xilinx-series-7 OS=centos OS_VERSION=6
+  - TOOLCHAIN=xilinx-series-7 OS=centos OS_VERSION=7
+  - TOOLCHAIN=xilinx-series-7 OS=centos OS_VERSION=8
 
-      script:
-            - tuttest README.md unnamed3 | bash -
-    - #
-      install:
-            - tuttest README.md unnamed2 | bash -
+install:
+ - if [ "$OS" = "ubuntu" ]; then export INSTALL_COMMAND="apt update && apt install -y"; fi
+ - if [ "$OS" = "centos" ]; then export INSTALL_COMMAND="yum -y install"; fi
+ - if [ "$OS" = "ubuntu" ]; then export INSTALL_PACKAGES="locales software-properties-common git wget xz-utils make build-essential zlib1g libz-dev libffi-dev ca-certificates"; fi
+ - if [ "$OS" = "centos" ]; then export INSTALL_PACKAGES="git wget make gcc-c++ zlib-devel xz libffi-devel perl-Pod-Html diffutils byacc"; fi
+ - export DOCKER_NAME=test
+ - export IN_DOCKER_EXEC="docker exec -it $DOCKER_NAME"
+ - docker create
+        --name $DOCKER_NAME
+        --env LANG="en_US.UTF-8"
+        --env LANGUAGE="en_US"
+        --env DEBIAN_FRONTEND="noninteractive"
+        -v /home/travis:/home/travis
+        --workdir $PWD
+        --interactive
+        --tty
+        $OS:$OS_VERSION
+ - docker start $DOCKER_NAME
+ - $IN_DOCKER_EXEC bash -c "$INSTALL_COMMAND $INSTALL_PACKAGES"
+ - if [ "$OS" = "ubuntu" ]; then $IN_DOCKER_EXEC locale-gen en_US.UTF-8; fi
+ - if [ "$OS" = "ubuntu" ]; then $IN_DOCKER_EXEC update-locale; fi
```

Comment:
What is going on with the locale stuff here?

Reply:
It was added to get rid of missing/unset locale warning when installing packages using apt. It is not necessary and can be removed, but we’ll have a bunch of warnings in CI log